plugin/cog/pom.xml

```diff
@@ -0,0 +1,92 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<project xmlns="http://maven.apache.org/POM/4.0.0"
+         xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+         xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+    <parent>
+        <artifactId>imageio-ext-plugin</artifactId>
+        <groupId>it.geosolutions.imageio-ext</groupId>
+        <version>1.3-SNAPSHOT</version>
+    </parent>
+    <modelVersion>4.0.0</modelVersion>
+
+    <artifactId>imageio-ext-cog</artifactId>
+    <packaging>pom</packaging>
+    <name>cog</name>
+
+    <properties>
+        <azure.version>12.0.0-preview.3</azure.version>
+        <okhttp.version>4.2.0</okhttp.version>
+        <aws.version>2.9.2</aws.version>
+        <ehcache.version>3.4.0</ehcache.version>
+    </properties>
+
+    <modules>
+        <module>cog-reader</module>
+        <module>cog-commons</module>
+        <module>cog-streams</module>
+        <module>cog-rangereader-http</module>
+        <module>cog-rangereader-s3</module>
+        <module>cog-rangereader-azure</module>
+    </modules>
+
+    <dependencyManagement>
+        <dependencies>
+            <dependency>
+                <groupId>it.geosolutions.imageio-ext</groupId>
+                <artifactId>imageio-ext-tiff</artifactId>
+                <version>${project.version}</version>
+            </dependency>
+            <dependency>
+                <groupId>it.geosolutions.imageio-ext</groupId>
+                <artifactId>imageio-ext-cog-commons</artifactId>
+                <version>${project.version}</version>
+            </dependency>
+            <dependency>
+                <groupId>it.geosolutions.imageio-ext</groupId>
+                <artifactId>imageio-ext-cog-rangereader-http</artifactId>
+                <version>${project.version}</version>
+            </dependency>
+            <dependency>
+                <groupId>it.geosolutions.imageio-ext</groupId>
+                <artifactId>imageio-ext-cog-rangereader-s3</artifactId>
+                <version>${project.version}</version>
+            </dependency>
+            <dependency>
+                <groupId>it.geosolutions.imageio-ext</groupId>
+                <artifactId>imageio-ext-cog-rangereader-azure</artifactId>
+                <version>${project.version}</version>
+            </dependency>
+            <dependency>
+                <groupId>it.geosolutions.imageio-ext</groupId>
+                <artifactId>imageio-ext-cog-streams</artifactId>
+                <version>${project.version}</version>
+            </dependency>
+            <dependency>
+                <groupId>com.azure</groupId>
+                <artifactId>azure-storage-blob</artifactId>
+                <version>${azure.version}</version>
+            </dependency>
+            <dependency>
+                <groupId>com.squareup.okhttp3</groupId>
```

Comment:
OkHttp seems to be a pretty popular client, but I'm not tied to it.  This was actually a rewrite as I originally used the Java 11 HttpClient, but Torben pointed out that this needs to be compatible with Java 8, so I moved to OkHttp.  I have no issue moving to commons-httpclient as it seems to support async connections.  My question now is, would you be against introducing v5 of the client?  It supports http/2 multiplexing, which eliminates the concerns of how many connections get created.  This seems like the best modern approach to take, but still introduces a new dependency.  Or potentially I could provide an http1 and http2 client?

https://hc.apache.org/httpcomponents-client-5.0.x/httpclient5/examples/AsyncClientH2Multiplexing.java

Reply:
If it's not aligned with the GeoTools current usage it would be one more dependency nevertheless, at that point, probably better to keep it as is and concentrate on the other issues.

Reply:
Left it alone for now.  I'm fine with rewriting this in the future or refactoring this to be the http2 module and providing another http module based on commons-httpclient.